src/workerd/server/autogate.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2017-2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+#pragma once
+
+#include "kj/debug.h"
+#include <kj/map.h>
+#include <workerd/server/workerd.capnp.h>
+
+namespace workerd::server {
+
+// This class allows code changes to be rolled out independent of full binary releases. It enables
+// specific code paths to be gradually rolled out via our internal tooling.
+// See the equivalent file in our internal repo for more details.
+//
+// Workerd-specific gates can be added here.
+//
+// When making changes here, ensure you align them with autogate.h in workerd.
```

Comment:
Would you mind expanding the documentation to show an example of how a gate is to be used in the source?

Reply:
It's there in the internal docs, but I will add it here too.